app/templates/ui/app/login/login.service.js

```diff
@@ -117,20 +154,24 @@
         _toStateName = next.name;
         _toStateParams = nextParams;
       }
-      if (!isAuthenticated() && routeIsProtected(next.name)) {
-        event.preventDefault();
-        loginPrompt();
-        if (_loginMode !== 'full') {
-          if (deregisterLoginSuccess) {
-            deregisterLoginSuccess();
-            deregisterLoginSuccess = null;
-          }
-          deregisterLoginSuccess = $rootScope.$on('loginService:login-success', function() {
-            deregisterLoginSuccess();
-            deregisterLoginSuccess = null;
-            $state.go(next.name, nextParams);
+
+      if (routeIsProtected(next.name)) {
+        var auth = getAuthenticatedStatus();
+
+        if (angular.isFunction(auth.then)) {
+          auth.then(function() {
+            if (!isAuthenticated()) {
+              //this does NOT block requests in a timely fashion...
```

Comment:
Got it working so the UI seems correct to the user however it still pushes out requests a bit fast. may need to tweak block route or the login interceptor?